scripts/srt-dev.lua

```diff
@@ -0,0 +1,887 @@
+fields.smoother_block = ProtoField.none("srt_dev.smoother_block", "Smoother Block", base.NONE)
+
+-- Handshake packet, ext_type == SRT_CMD_SID filed
+fields.sid_block = ProtoField.none("srt_dev.sid_block", "SID Block", base.NONE)
+
+-- SRT flags
+local flag_state_select = {
+	[0] = "Not set",
+	[1] = "Set"
+}
+fields.srt_opt_tsbpdsnd = ProtoField.uint32("srt_dev.srt_opt_tsbpdsnd", "SRT_OPT_TSBPDSND", base.HEX, flag_state_select, 0x1)
+fields.srt_opt_tsbpdrcv = ProtoField.uint32("srt_dev.srt_opt_tsbpdrcv", "SRT_OPT_TSBPDRCV", base.HEX, flag_state_select, 0x2)
+fields.srt_opt_haicrypt = ProtoField.uint32("srt_dev.srt_opt_haicrypt", "SRT_OPT_HAICRYPT", base.HEX, flag_state_select, 0x4)
+fields.srt_opt_tlpktdrop = ProtoField.uint32("srt_dev.srt_opt_tlpktdrop", "SRT_OPT_TLPKTDROP", base.HEX, flag_state_select, 0x8)
+fields.srt_opt_nakreport = ProtoField.uint32("srt_dev.srt_opt_nakreport", "SRT_OPT_NAKREPORT", base.HEX, flag_state_select, 0x10)
+fields.srt_opt_rexmitflg = ProtoField.uint32("srt_dev.srt_opt_rexmitflg", "SRT_OPT_REXMITFLG", base.HEX, flag_state_select, 0x20)
+fields.srt_opt_stream = ProtoField.uint32("srt_dev.srt_opt_stream", "SRT_OPT_STREAM", base.HEX, flag_state_select, 0x40)
+
+-- ACK fileds
+fields.last_ack_pack = ProtoField.uint32("srt_dev.last_ack_pack", "Last ACK Packet Sequence Number", base.DEC)
+fields.rtt = ProtoField.uint32("srt_dev.rtt", "Round Trip Time", base.DEC)
+fields.rtt_variance = ProtoField.uint32("srt_dev.rtt_variance", "Round Trip Time Variance", base.DEC)
+fields.buf_size = ProtoField.uint32("srt_dev.buf_size", "Available Buffer Size", base.DEC)
+fields.pack_rcv_rate = ProtoField.uint32("srt_dev.pack_rcv_rate", "Packet Receiving Rate", base.DEC)
+fields.est_link_capacity = ProtoField.uint32("srt_dev.est_link_capacity", "Estimated Link Capacity", base.DEC)
+fields.rcv_rate = ProtoField.uint32("srt_dev.rcv_rate", "Receiving Rate", base.DEC)
+
+-- ACKACK fileds
+fields.ack_seq_num = ProtoField.uint32("srt_dev.ack_seq_num", "ACK sequence number", base.DEC)
+fields.ctl_info = ProtoField.uint32("srt_dev.ctl_info", "Control Information", base.DEC)
+
+-- KMRSP fileds
+local srt_km_state_select = {
+	[0] = "[SRT_KM_UNSECURED]",
+	[1] = "[SRT_KM_SECURING]",
+	[2] = "[SRT_KM_SECURED]",
+	[3] = "[SRT_KM_NOSECRET]",
+	[4] = "[SRT_KM_BADSECRET]"
+}
+fields.km_err = ProtoField.uint32("srt_dev.km_err", "Key Message Error", base.HEX, srt_km_state_select, 0xF)
+
+-- NAK Control Packet fileds
+fields.lost_list_tree = ProtoField.none("srt_dev.lost_list_tree", "Lost Packet List", base.NONE)
+fields.lost_pack_seq = ProtoField.uint32("srt_dev.lost_pack_seq", "Lost Packet Sequence Number", base.DEC)
+fields.lost_pack_range_tree = ProtoField.none("srt_dev.lost_pack_range_tree", "Lost Packet Range", base.NONE)
+fields.lost_start = ProtoField.uint32("srt_dev.lost_start", "Lost Starting Sequence", base.DEC)
+fields.lost_up_to = ProtoField.uint32("srt_dev.lost_up_to", "Lost Up To(including)", base.DEC)
```

Comment:
Not sure if this is properly done, just keep in mind: the most significant bit in the sequence numbers here is set if it marks the beginning of the range, then the next number in the array is an end of range. Otherwise it's a single loss. Any number of such declarations can be provided in this message.